Backend/marketuserpost.js

```diff
@@ -54,6 +54,30 @@ async function marketcreatePost(req, res) {
     }
 }
 
+
+/**
+ * Function to retrieve market posts and sort them by createdAt 
+ */
+async function retrieveMarketPosts(req, res) {
+    try {
+        const marketPosts = await Marketpost.find().sort({ createdAt: 'desc' });
```

Comment:
better to break it into parts instead of keep await the mongo query

Reply:
the function is divided.